### PR TITLE
(PC-16884) test(Web): enable service worker test in GitHub actions

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-16884

## Todo

- [ ] Remove unecessary repository https://github.com/pass-culture/puppeteer-headful
- [x] See with @apibrac if `kopax/puppeteer-headful:latest` can be used (this is the [image](https://github.com/mujo-code/puppeteer-headful))
- [x] See if we need to improve [copy](https://github.com/pass-culture/pass-culture-app-native/blob/PC-16884/scripts/service-worker-test-ci.mjs#L78) step of 2nd version (see below)

Regarding the 3rd point, we must keep puppeteer after opening the 1st build (versionFrom) then use the 2nd build (versionTo), otherwise if we split the js, it will be like testing two new app with different browser, so without testing the service worker upgrade.

In order to observe the test sequences entirely in github action (so the versionTo copy to serve directory), I will use for the moment a comment to say that 2nd build will be use during the script because of sequence requirement necessary for the test.

It's maybe overkill, but there is another way we can do to have the most easiest to maintain test (which I will do in an improvement ticket if requested), we could improve to see the whole sequence of the test in GitHub actions by doing as follow:

- Remove basic http server
- Use express in the script
- Run the express server in background and open browser on startup
- Use github action and newly created endpoints to trigger test steps

## Checklist

I have:

- [x] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.

### CODE REVIEW

**Review by cgaunet — APPROVED** [review]

**Review by apibrac — APPROVED** [review]
Il faut juste mettre un repo pass Culture. On ne decrait pas utiliser un repo perso comme ça

File: .github/workflows/service-worker-test.yml [hunk 1]
```diff
@@ -12,22 +12,98 @@ on:
...
-        env:
-          PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: 'true'
+      # Docker entrypoint is otherwise ignored in GitHub Action
+      - name: Docker entrypoint
+        shell: bash
+        run: /entrypoint.sh
+
+      # Useful to get the latest service worker test script
+      - name: Checkout registry
+        uses: actions/checkout@v3
```
[the remark this answers is not in the document]
Comment (kopax-polyconseil) [reply .github/workflows/service-worker-test.yml:33]: Car le script ci n'est pas dans les vieilles version, il est que dans cette branche (pour le moment)

**Comment by kopax-polyconseil** [comment]
Ok je regarde lundi pour héberger notre image et je marge ensuite! Bon WE!

File: .github/workflows/service-worker-test.yml [hunk 2]
```diff
@@ -12,22 +12,98 @@ on:
...
 jobs:
-  installDependencies:
-    name: Install Dependencies
+  testServiceWorker:
+    name: Test service worker
     runs-on: ubuntu-latest
+    env:
+      PORT: 5000
+    container:
+      image: kopax/puppeteer-headful:latest
```
Comment (kopax-polyconseil) [inline .github/workflows/service-worker-test.yml:23]: Todo: replace with [europe-west1-docker.pkg.dev/passculture-infra-prod/pass-culture-tools/puppeteer-headful:node18](http://europe-west1-docker.pkg.dev/passculture-infra-prod/pass-culture-tools/puppeteer-headful:node18)

**Pull request merged**